disputer/disputer.js

```diff
@@ -71,37 +73,52 @@ class Disputer {
           at: "Disputer",
           message: "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute.",
           id: disputeableLiquidation.id,
-          sponsor: disputeableLiquidation.sponsor
-        });
-
-        Logger.debug({
-          at: "Disputer",
-          message: "Dispute call error message",
-          id: disputeableLiquidation.id,
           sponsor: disputeableLiquidation.sponsor,
           error: error
         });
         continue;
       }
 
-      // TODO: handle transaction failures.
-      const receipt = await dispute.send({
-        from: this.account,
-        gas: 1500000,
-        gasPrice: this.gasEstimator.getCurrentFastPrice()
-      });
+      disputePromises.push(
```

Comment:
Does structuring it this way break the inventory management above? Let's say the both disputes cost 1 ETH and the dispute bot only has 1 ETH. Won't it try to send both transactions without realizing that one will definitely fail (because the first will spend the 1 ETH)?

Reply:
Hm this is a good point I hadn't thought of. @chrismaree why did you originally decide to call all of the liquidations in `liquidator` via `Promise.all`?

Reply:
Talked with Chris offline, he had not thought of this issue before. I will make both bots consistent and prefer the iterative approach where we resolve promises 1 by 1.

Reply:
Done, both liquidator and disputer resolve promises one by one